src/shared/sol-iio.h

```diff
@@ -0,0 +1,141 @@
+ *     notice, this list of conditions and the following disclaimer in
+ *     the documentation and/or other materials provided with the
+ *     distribution.
+ *   * Neither the name of Intel Corporation nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <sol-buffer.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @file
+ * @brief Routines to access Linux Industrial I/O (iio) devices under Solleta
+ */
+
+struct sol_iio_device;
+struct sol_iio_channel;
+
+struct sol_iio_config {
+#define SOL_IIO_CONFIG_API_VERSION (1)
+    uint16_t api_version;
+    const char *trigger_name; /**< Name of IIO trigger to be used on this device. If NULL or empty, will try to use device current trigger. If device has no current trigger, will create a 'sysfs_trigger' and use it. */
+    void (*sol_iio_reader_cb)(void *data, struct sol_iio_device *device); /**< Callback to be called when get new device readings on buffer */
+    const void *data; /**< User defined data to be sent to sol_iio_reader_cb */
+    int buffer_size; /**< size of reading buffer. 0: use device default; -1: disable buffer and readings will be performed on channel files on sysfs. */
+    int sampling_frequency; /**< Device sampling frequency. -1 uses device default */
+};
+
+struct sol_iio_channel_config {
+#define SOL_IIO_CHANNEL_CONFIG_API_VERSION (1)
+    uint16_t api_version;
+    double scale; /**< Channel scale, to be applied to raw readings. -1 uses device default. Some devices share scale among all channels, so changing one will change all. If, in this case, different channels set different scales the result is unknown. */
+    bool use_custom_offset; /**< If true, will use user defined offset on member #offset of this struct */
+    int offset; /**< Channel offset, to be added to raw readings. Some devices share offset among all channels, so changing one will change all. If, in this case, different channels set different offsets the result is unknown. */
+};
+
+#define SOL_IIO_CHANNEL_CONFIG_INIT { .scale = -1.0, .use_custom_offset = false }
```

Comment:
This check is to link/sync between user version and library version. Then you provide a number you got at compile time (user version), distribute the binary. The libsoletta may be updated in the system and your binary not, in that case these numbers will get out-of-sync and your check will raise an issue.

Then, yes, you should put the .api_version = SOL_IIO_CHANNEL_CONFIG_API_VERSION in that macro.